src/XRP/xrp-payment.ts

```diff
@@ -0,0 +1,68 @@
+import { Payment } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import XRPCurrencyAmount from './xrp-currency-amount'
+import XRPPath from './xrp-path'
+
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPPayment {
+  public static from(payment: Payment): XRPPayment | undefined {
+    const paymentAmountValue = payment.getAmount()?.getValue()
+    const amount =
+      paymentAmountValue && XRPCurrencyAmount.from(paymentAmountValue)
+    if (!amount) {
+      return undefined // amount is required
+    }
```

Comment:
@keefertaylor might hate this, but you _can_ do
```
if (!amount) return undefined
```

Some people really hate the 1-line `if return` style though, so this isn't necessarily a recommendation. Not sure what the general code style of this repo is.

Reply:
the linter is making me do it like this so I'll just leave it consistent